src/libraries/Microsoft.PowerFx.Core/Binding/INameResolver.cs

```diff
@@ -36,6 +38,8 @@ internal interface INameResolver
 
         IEnumerable<TexlFunction> Functions { get; }
 
+        IReadOnlyDictionary<string, IGlobalSymbol> GlobalSymbols { get; }
```

Comment:
>IReadOnlyDictionary [](http://example.com/codeflow?start=8&length=19)

Consider making this an IEnumerable<>
Because we already do lookup via Lookup() on line 50.

Reply:
along those lines - do we need a new IGlobalSymbol? or can we use existing NameLookupInfo that is used elsewhere on this interface?

Ie, consider the duality:

var sym = resolver.GlobalSymbols["MyVar"]

resolver.Lookup(myVar, out var sym);

Reply:
We probably need to discuss that one